channel.go

```diff
@@ -52,28 +49,26 @@ type Channel struct {
 	LastMessageAt time.Time `json:"last_message_at"`
 
 	client *Client
+	lock   *sync.Mutex
 }
 
-type channelQueryResponse struct {
+type channelResponse struct {
 	Channel  *Channel         `json:"channel,omitempty"`
 	Messages []*Message       `json:"messages,omitempty"`
 	Members  []*ChannelMember `json:"members,omitempty"`
 	Read     []*ChannelRead   `json:"read,omitempty"`
 }
 
-func (ch *Channel) update(q channelQueryResponse) {
-	// atomically add or create new lock in channel lock store
-	lock, _ := channelsLock.LoadOrStore(ch.CID, &sync.Mutex{})
-
-	lock.(*sync.Mutex).Lock()
-	defer lock.(*sync.Mutex).Unlock()
+func (ch *Channel) update(q channelResponse) {
```

Comment:
Not an update; apply(Response).

Reply:
`ch.update` sounds more logical then `ch.apply`

Reply:
Update is what you receive; apply is what the fuction does. So applyResponse makes clear the response is applied to ch. Update can mean anything, like change q, mention the presense of an update or whatever. Not that important though. Up to you.